exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -4,157 +4,154 @@
-	    purses[howManyPlayers()] = 0;
-	    inPenaltyBox[howManyPlayers()] = false;
+	public void addPlayer(Player player) {
+	    players.add(player);
+		int numOfPlayers = howManyPlayers();
+
+	    inPenaltyBox[numOfPlayers] = false;
 	    
-	    System.out.println(playerName + " was added");
+	    System.out.println(player.getName() + " was added");
 	    System.out.println("They are player number " + players.size());
-		return true;
 	}
 	
-	public int howManyPlayers() {
+	private int howManyPlayers() {
 		return players.size();
 	}
 
-	public void roll(int roll) {
-		System.out.println(players.get(currentPlayer) + " is the current player");
-		System.out.println("They have rolled a " + roll);
+	public void moveOnRoll(int rollValue) {
+		System.out.println(players.get(currentPlayer).getName() + " is the current player");
+		System.out.println("They have rolled a " + rollValue);
 		
 		if (inPenaltyBox[currentPlayer]) {
-			if (roll % 2 != 0) {
-				isGettingOutOfPenaltyBox = true;
-				
-				System.out.println(players.get(currentPlayer) + " is getting out of the penalty box");
-				places[currentPlayer] = places[currentPlayer] + roll;
-				if (places[currentPlayer] > 11) places[currentPlayer] = places[currentPlayer] - 12;
-				
-				System.out.println(players.get(currentPlayer) 
-						+ "'s new location is " 
-						+ places[currentPlayer]);
-				System.out.println("The category is " + currentCategory());
+			isGettingOutOfPenaltyBox = rollValue % 2 != 0;
+
+			if (isGettingOutOfPenaltyBox) {
+				System.out.println(players.get(currentPlayer).getName() + " is getting out of the penalty box");
+
+				movePlayer(rollValue);
 				askQuestion();
-			} else {
-				System.out.println(players.get(currentPlayer) + " is not getting out of the penalty box");
-				isGettingOutOfPenaltyBox = false;
-				}
+
+				return;
+			}
+			System.out.println(players.get(currentPlayer).getName() + " is not getting out of the penalty box");
+			isGettingOutOfPenaltyBox = false;
 			
 		} else {
-		
-			places[currentPlayer] = places[currentPlayer] + roll;
-			if (places[currentPlayer] > 11) places[currentPlayer] = places[currentPlayer] - 12;
-			
-			System.out.println(players.get(currentPlayer) 
-					+ "'s new location is " 
-					+ places[currentPlayer]);
-			System.out.println("The category is " + currentCategory());
+			movePlayer(rollValue);
 			askQuestion();
 		}
 		
 	}
 
+	private void movePlayer(int rollValue) {
+    	Player theCurrentPlayer = players.get(currentPlayer);
+		theCurrentPlayer.setPlaceOnBoard(rollValue);
+		if (theCurrentPlayer.getPlaceOnBoard() > numOfPlaces - 1) {
+			theCurrentPlayer.setPlaceOnBoard(-numOfPlaces);
+		}
+
+		System.out.println(theCurrentPlayer.getName()
+				+ "'s new location is "
+				+ theCurrentPlayer.getPlaceOnBoard());
+		System.out.println("The category is " + currentCategory());
+	}
+
 	private void askQuestion() {
-		if (currentCategory() == "Pop")
+		if (currentCategory().equals("Pop"))

```

Comment:
I think this could be simplified by using a container for the question categories.